File: src/main/java/com/verlumen/tradestream/backtesting/BacktestingModule.java
package com.verlumen.tradestream.backtesting;

import com.google.inject.AbstractModule;

public final class BacktestingModule extends AbstractModule {
  public static BacktestingModule create() {
    return new BacktestingModule();
  }

  @Override
  protected void configure() {
    bind(BacktestRequestFactory.class).to(BacktestRequestFactoryImpl.class);
    bind(BacktestRunner.class).to(BacktestRunnerImpl.class);
    bind(FitnessCalculator.class).to(FitnessCalculatorImpl.class);
    bind(GAEngineFactory.class).to(GAEngineFactoryImpl.class);
    bind(GeneticAlgorithmOrchestrator.class).to(GeneticAlgorithmOrchestratorImpl.class);
  }
}

File: src/main/java/com/verlumen/tradestream/backtesting/BacktestingModule.kt
package com.verlumen.tradestream.backtesting

import com.google.inject.AbstractModule

class BacktestingModule : AbstractModule() {
    override fun configure() {
        bind(BacktestRequestFactory::class.java).to(BacktestRequestFactoryImpl::class.java)
        bind(BacktestRunner::class.java).to(BacktestRunnerImpl::class.java)

        // Walk-forward validation components
        bind(WalkForwardRunner::class.java)
        bind(StrategyValidator::class.java).toInstance(StrategyValidator())
    }
}
